store/copr/mpp.go

```diff
@@ -231,6 +237,34 @@ func (m *mppIterator) handleDispatchReq(ctx context.Context, bo *tikv.Backoffer,
 	m.establishMPPConns(bo, req, taskMeta)
 }
 
+// NOTE: We do not retry here, because retry is helpless when errors result from TiFlash or Network. If errors occur, the execution will stop after some minutes.
+func (m *mppIterator) cancelMppTasks(bo *tikv.Backoffer, meta *mpp.TaskMeta) {
+	killReq := &mpp.CancelTaskRequest{
+		Meta: meta,
+	}
+
+	wrappedReq := tikvrpc.NewRequest(tikvrpc.CmdMPPCancel, killReq, kvrpcpb.Context{})
+	wrappedReq.StoreTp = kv.TiFlash
+
+	usedStoreAddrs := make(map[string]bool)
+	for _, task := range m.tasks {
+		// get the store address of running tasks
+		if task.State != kv.MppTaskReady && !usedStoreAddrs[task.Meta.GetAddress()] {
```

Comment:
I think the task should be cancelled iff the state is running.

Reply:
done.